entity-web/src/main/java/eu/europeana/entity/web/controller/BaseRest.java

```diff
@@ -228,6 +229,28 @@ protected String extractSolrFieldName(String query) {
 	return fieldName;
     }
 
+	/**
+	 * Validate language parameter
+	 * @param language
+	 * @throws ParamValidationException
+	 */
+	protected void validateLanguage(String language) throws ParamValidationException {
+	if (StringUtils.isEmpty(language))	{
+		return;
+	}
+	// multiple language not supported
+	if (StringUtils.contains(language, WebEntityConstants.COMMA)) {
+		throw new ParamValidationException(I18nConstants.UNSUPPORTED_MULTIPLE_LANG_VALUE,
+				CommonApiConstants.QUERY_PARAM_LANG, language);
+	}
+	// language value can be 'all' Or ISO language only
+	if (!StringUtils.equals(language, WebEntityConstants.PARAM_LANGUAGE_ALL) &&
+	     Arrays.stream(Locale.getISOLanguages()).noneMatch(isoLang -> StringUtils.equals(language, isoLang))){
```

Comment:
I think we should improve the performance by implementing validation like in Entity Management
By using a static set and use the contains method
see:
 private static final Set<String> ISO_LANGUAGES = Set.of(Locale.getISOLanguages());

https://github.com/europeana/entity-management/blob/6deb5764b59f0a481547ec0d8ca10ca6fd8d3b3a/entity-management-definitions/src/main/java/eu/europeana/entitymanagement/normalization/EntityFieldsCleaner.java#L309

Reply:
done